### DISCUSSION TITLE
Support searching for numeric literals with digit separators in GitHub Code Search

### DISCUSSION THREAD

**Discussion by tristaninf** [community/community#129896] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello GitHub Team,

I am writing to request an enhancement to the GitHub Code Search functionality. Specifically, I would like to see support for searching numeric literals with digit separators (underscores) in addition to the standard numeric format.

**Current Limitation:**

Currently, when searching for a numeric value such as `999999`, it is not possible to find matches that include digit separators like `999_999`. This feature would be extremely helpful for developers who utilize digit separators for better readability in their code, as it would allow us to locate all relevant instances of a numeric value regardless of formatting.

**Feature Request:**

I propose that GitHub Code Search be enhanced to:
1. Recognize numeric literals with digit separators (e.g., `999_999`) as equivalent to their standard numeric counterparts (e.g., `999999`).
2. Allow for searching both formats simultaneously using a consistent search pattern or an enhanced regular expression feature.

**Benefits:**

- Improved code search accuracy and efficiency for repositories that utilize digit separators for readability.
- Easier maintenance and code review processes by ensuring all instances of a numeric value can be identified, regardless of formatting.
- Enhanced developer experience by aligning search capabilities with modern coding standards and practices.

**Example:**

A search for `999999` should also return results for `999_999`, `99_99_99`, etc. Additionally, it would be beneficial if regular expressions could be used to specify patterns that match numeric values with optional digit separators.

Thank you for considering this enhancement. I believe it would significantly improve the usability of GitHub Code Search for many developers.

**Comment by look** [comment, marked as answer]
Thanks for the suggestion. I recommend using a regular expression for this use case.

For example, to search for variations of 1,000,000 you could use: [`/1[_,0]{6,}/`](https://github.com/search?q=%2F1%5B_%2C0%5D%7B6%2C%7D%2F&type=code). This isn't _exactly_ what you want: it will match `100_000` since it has 6 characters after the `1` just like `1000000` does. However, it will narrow the search space considerably. If there's a format regularly used in your codebase, you could provide a stricter regular expression.
